src/app.js

```diff
@@ -94,6 +94,27 @@ app.get('/publish-post', ensureAuthenticated, async function (req, res, next) {
 
 app.post('/publish-post', ensureAuthenticated, require('./publish-post'));
 
+app.get('/profile-posts', ensureAuthenticated, async function (req, res, next) {
+  const providerName = req.query.provider || 'facebook';
+  let profiles = [];
+  try {
+    const accessToken = getAccessTokenByProviderName(providerName, req);
+    profiles = await getProfilesForPublishing(providerName, accessToken);
+  } catch (err) {
+    console.error(err);
+    next(err);
+    return;
+  }
+  res.render('profile-posts', {
+    user: req.user,
+    providerName,
+    profiles,
+    success: undefined,
+  });
+});
+
+app.post('/profile-posts', ensureAuthenticated, require('./profile-posts'));
```

Comment:
Instead of displaying it under POST submission, I suggest adding it as a query parameter to GET /profile-posts, e.g. `/profile-posts?provider=facebook&profileId=1234`

Reply:
fixed